modules/generator/processor/localblocks/processor.go

```diff
@@ -310,6 +365,24 @@ func (p *Processor) completeBlock() error {
 	return nil
 }
 
+func (p *Processor) flushBlock(id uuid.UUID) error {
+	p.blocksMtx.RLock()
+	completeBlock := p.completeBlocks[id]
+	p.blocksMtx.RUnlock()
+
+	if completeBlock == nil {
+		return nil
+	}
+
+	ctx := context.Background()
+	err := p.writer.WriteBlock(ctx, completeBlock)
+	if err != nil {
+		return err
+	}
+	metricFlushedBlocks.WithLabelValues(p.tenant).Inc()
+	return nil
```

Comment:
As discussed offline, let's add a log message with the tenant and block ID.  level=info seems better than debug, and log volume isn't a concern.

Reply:
Added!